### PR TITLE
sbt 1.0.4. cleanup build settings

### PR DESCRIPTION
**Pull request by xuwei-k**

### CODE REVIEW

File: project/plugins.sbt [hunk 1]
```diff
@@ -1,7 +1,5 @@
 addSbtPlugin("com.typesafe"     % "sbt-mima-plugin"      % "0.1.18")
-addSbtPlugin("com.typesafe.sbt" % "sbt-javaversioncheck" % "0.1.0")
 addSbtPlugin("com.eed3si9n"     % "sbt-buildinfo"        % "0.7.0")
-addSbtPlugin("com.typesafe.sbt" % "sbt-start-script"     % "0.10.0")
```
Comment (xuwei-k) [inline project/plugins.sbt]: sbt-start-script and sbt-javaversioncheck does not available for sbt 1.0

File: publish.sh [hunk 2]
```diff
@@ -1,4 +0,0 @@
-#!/bin/bash
-
-sbt ++2.10.6 javaVersionCheck publishSigned \
-    ++2.11.11 javaVersionCheck publishSigned
```
Comment (xuwei-k) [inline publish.sh]: json4s 3.6 dropped Java 7 support. https://github.com/json4s/json4s/commit/43ebd6bbf2adf9ee23a7cb73946b6719b4f5249e
We can/should use just `sbt +publishSigned`

File: project/Dependencies.scala [hunk 3]
```diff
@@ -13,11 +13,12 @@ object Dependencies {
   lazy val scalaz_core  = "org.scalaz"                   %% "scalaz-core"          % "7.2.14"
   lazy val paranamer    = "com.thoughtworks.paranamer"   %  "paranamer"            % "2.8"
   lazy val specs        = Def.setting{
-    // TODO enable if specs2 for Scala 2.13.0-M2 released
-    PartialFunction.condOpt(CrossVersion.partialVersion(scalaVersion.value)){
-      case Some((2, v)) if v <= 12 =>
+    CrossVersion.partialVersion(scalaVersion.value) match {
+      case Some((2, v)) if v <= 10 =>
         Seq("org.specs2" %% "specs2-scalacheck" % "3.9.4" % "test")
```
Comment (xuwei-k) [inline project/Dependencies.scala]: specs2 4.x does not support Scala 2.10

**Comment by seratch** [comment]
Thanks, LGTM.

**Pull request merged**